UnitTests/Views/TreeTableSourceTests.cs

```diff
@@ -0,0 +1,287 @@
+││ └─DeLorean     │Time travelling car │
+│└+Route 66       │Great race course   │
+";
+
+		TestHelpers.AssertDriverContentsAre (expected, output);
+
+		// when pressing left we should collapse the top route again
+		Application.Top.ProcessHotKey (new KeyEvent (Key.CursorLeft, new KeyModifiers ()));
+
+
+		tv.Draw ();
+
+		expected =
+			@"
+│Name          │Description            │
+├──────────────┼───────────────────────┤
+│├+Lost Highway│Exciting night road    │
+│└+Route 66    │Great race course      │
+";
+
+		TestHelpers.AssertDriverContentsAre (expected, output);
+	}
+
+	[Fact, AutoInitShutdown]
+	public void TestTreeTableSource_BasicExpanding_WithMouse ()
+	{
+		var tv = GetTreeTable (out _);
+
+		tv.Style.GetOrCreateColumnStyle (1).MinAcceptableWidth = 1;
+
+		tv.Draw ();
+
+		string expected =
+			@"
+│Name          │Description            │
+├──────────────┼───────────────────────┤
+│├+Lost Highway│Exciting night road    │
+│└+Route 66    │Great race course      │";
+
+		TestHelpers.AssertDriverContentsAre (expected, output);
+
+		Assert.Equal (2, tv.Table.Rows);
+
+		// top left is selected cell
+		Assert.Equal (0, tv.SelectedRow);
+		Assert.Equal (0, tv.SelectedColumn);
+
+		Assert.True(tv.OnMouseEvent (new MouseEvent () { X = 2,Y=2,Flags = MouseFlags.Button1Clicked}));
+			
+		tv.Draw ();
+
+		expected =
+			@"
+│Name             │Description         │
+├─────────────────┼────────────────────┤
+│├-Lost Highway   │Exciting night road │
+││ ├─Ford Trans-Am│Talking thunderbird │
+││ └─DeLorean     │Time travelling car │
+│└+Route 66       │Great race course   │
+";
+
+		TestHelpers.AssertDriverContentsAre (expected, output);
+
+		// Clicking to the right/left of the expand/collapse does nothing
+		tv.OnMouseEvent (new MouseEvent () { X = 3, Y = 2, Flags = MouseFlags.Button1Clicked });
+		tv.Draw ();
+		TestHelpers.AssertDriverContentsAre (expected, output);
+		tv.OnMouseEvent (new MouseEvent () { X = 1, Y = 2, Flags = MouseFlags.Button1Clicked });
+		tv.Draw ();
+		TestHelpers.AssertDriverContentsAre (expected, output);
+
+		// Clicking on the + again should collapse
+		tv.OnMouseEvent (new MouseEvent () { X = 2, Y = 2, Flags = MouseFlags.Button1Clicked });
+		tv.Draw ();
+		
+		expected =
+			@"
+│Name          │Description            │
+├──────────────┼───────────────────────┤
+│├+Lost Highway│Exciting night road    │
+│└+Route 66    │Great race course      │";
+
+		TestHelpers.AssertDriverContentsAre (expected, output);
+
+	}
+
+	[Fact, AutoInitShutdown]
+	public void TestTreeTableSource_CombinedWithCheckboxes ()
+	{
+		var tv = GetTreeTable (out var treeSource);
+
+		CheckBoxTableSourceWrapperByIndex checkSource;
+		tv.Table = checkSource = new CheckBoxTableSourceWrapperByIndex (tv, tv.Table);
+		tv.Style.GetOrCreateColumnStyle (2).MinAcceptableWidth = 1;
+
+		tv.Draw ();
+
+		string expected =
+			@"
+    │ │Name          │Description          │
+├─┼──────────────┼─────────────────────┤
+│☐│├+Lost Highway│Exciting night road  │
```

Comment:
Is there a way to factor these tests to test what you're trying to test without relying on text output? Specifically, if the glyph for the check changes in the future, can these tests be made such that they don't need to be adapted?

Reply:
The test is to confirm that checkboxes play nicely with the tree source.

Since they are both `ITableSource` its important they don't break one another when used together.

The most comprehensive way of doing this is to check expanded/checked statuses but also confirm with draw check.  But I agree we don't want it to be style dependent.  

I have updated the test to explicitly set (and reset) the `ConfigurationManager` runes that should be used.:

https://github.com/gui-cs/Terminal.Gui/pull/2685/commits/3eff9d4d9829620c4d824a90450b7c47a3c75eba